bin.js

```diff
@@ -662,6 +662,12 @@ async function uploadData (uploadURL, authToken, filename, opts) {
 
   console.log(`Uploading data for ${filePrefix} and ${filePrefix}.html`)
 
+  // Check if path exists before uploading
+  if(!fs.existsSync(path.resolve(filePrefix))) {
+    console.error('File '+path.resolve(filePrefix)+' does not exist')
+    process.exit(1)
+  }
+
   const result = await tarAndUploadPromisified(path.resolve(filePrefix), uploadURL, authToken, { private: isPrivate })
```

Comment:
There's an inherent challenge with using `fs.existSync()` because while the file may not exist when `fs.existsSync()` is called, it may exist immediately after it's called ;-) 

If you look inside the `tarAndUploadPromisified()` implementation (https://github.com/nearform/node-clinic/blob/master/lib/tar-and-upload.js) you'll see that the stream object returned by `tar.pack()` does not have a proper error handler attached to it. That's really what you want to do here. That is, rather than checking to see if the file exists before attempting to tar it, go ahead and attempt and handle the error appropriately when it does not.

@mcollina is more familiar with these particular bits. Matteo: can I ask you to take a look here?

Reply:
@jasnell - I'm having some trouble reproducing this issue this morning. 
@mcollina 
Look inside - `function ontarred (err, zipped)`

This line seems to be handling the issue already? 
`if (empty) return cb(new Error('No data to upload'))`

If the tar is empty it won't attempt to upload and returns the error message instead.

Reply:
@mcollina - Any feedback on the above? ^

Reply:
I would replace `stream-collect` with some custom code and the use of `end-of-stream`. Between pumpify and `collect`, the error should be handled properly anyway, but better be sure.

Reply:
@mcollina - Alright. I'll work on implementing something like this today.

Reply:
@mcollina @jasnell 

> That is, rather than checking to see if the file exists before attempting to tar it, go ahead and attempt and handle the error appropriately when it does not.

Hopefully this is better.